### PR TITLE
Run docs on PR

### PR DESCRIPTION
**Pull request by Itxaka**
Only push on main branch, when files are changed.
On PR, always rebuild

Fixes https://github.com/rancher/elemental/issues/496

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

**Comment by kkaempf** [comment]
@nunix !

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
   push:
     branches:
       - main
+    paths:
+      - docs/**
+      - docusaurus.config.js
+      - package.json
+      - Vars.js
+      - sidebars.js
+      - yarn.lock
```
Comment (Itxaka) [inline .github/workflows/docs.yaml:14]: @nunix I think I got all the files here, is there anything else than can change that affects the docs?

File: yarn.lock [hunk 2]
```diff
@@ -3464,6 +3477,13 @@ cypress-file-upload@^5.0.8:
   resolved "https://registry.yarnpkg.com/cypress-file-upload/-/cypress-file-upload-5.0.8.tgz#d8824cbeaab798e44be8009769f9a6c9daa1b4a1"
   integrity sha512-+8VzNabRk3zG6x8f8BWArF/xA/W0VK4IZNx3MV0jFWrJS/qKn8eHfa5nU73P9fOQAgwHFJx7zjg4lwOnljMO8g==
 
+cypress-plugin-tab@^1.0.5:
```
Comment (Itxaka) [inline yarn.lock:3480]: This were missing when we re-introduced the deps for UI tests Im guessing.

**Review by mjura — APPROVED** [review]
LGTM

**Pull request merged**